lib/internal/test_runner/test.js

```diff
@@ -29,12 +31,27 @@ const kMultipleCallbackInvocations = 'multipleCallbackInvocations';
 const kParentAlreadyFinished = 'parentAlreadyFinished';
 const kSubtestsFailed = 'subtestsFailed';
 const kTestCodeFailure = 'testCodeFailure';
+const kTestTimeoutFailure = 'testTimeoutFailure';
 const kDefaultIndent = '    ';
 const noop = FunctionPrototype;
 const isTestRunner = getOptionValue('--test');
 const testOnlyFlag = !isTestRunner && getOptionValue('--test-only');
 // TODO(cjihrig): Use uv_available_parallelism() once it lands.
 const rootConcurrency = isTestRunner ? cpus().length : 1;
+const rootTimeout = 30_0000;
+
+
+function PromiseTimeout(promise, timeout) {
```

Comment:
Wondering if we can't use `AbortSignal.timeout` here

Reply:
interesting. how would you suggest doing that? it can be useful to pass a signal into the test context

Reply:
And if we pass an abortSignal to the callback - it should proabbly also be aborted here https://github.com/nodejs/node/blob/0fa7730d800b8ac072dda53ad37889bdbea376e3/lib/internal/test_runner/test.js
so `AbortSignal.timeout` wont cover that. WDYT?

Reply:
Yeah, a `signal` parameter to the test and then cancelling it to enable deep cancellation is probably a good idea but can also be done in a follow-up PR